wallet/src/main/scala/org/bitcoins/wallet/models/OutgoingTransactionTable.scala

```diff
@@ -0,0 +1,162 @@
+package org.bitcoins.wallet.models
+
+import org.bitcoins.core.crypto._
+import org.bitcoins.core.currency._
+import org.bitcoins.core.number.UInt32
+import org.bitcoins.core.protocol.script.EmptyScriptSignature
+import org.bitcoins.core.protocol.transaction._
+import org.bitcoins.core.wallet.fee.{FeeUnit, SatoshisPerVirtualByte}
+import slick.jdbc.SQLiteProfile.api._
+import slick.lifted.{PrimaryKey, ProvenShape}
+
+/** Represents a relevant transaction for the wallet that we should be keeping track of */
+case class OutgoingTransactionDb(
+    txIdBE: DoubleSha256DigestBE,
+    transaction: Transaction,
+    unsignedTxIdBE: DoubleSha256DigestBE,
+    unsignedTx: Transaction,
+    numInputs: Int,
+    numOutputs: Int,
+    lockTime: UInt32,
+    txFee: CurrencyUnit,
+    expectedFeeRate: FeeUnit,
+    actualFeeRate: FeeUnit) {
```

Comment:
I think we should add `wtxid: Option[DoubleSha256DigestBE` and `tx_type` here, also I think we should have a field called `outgoingAmount` which indicates how much money is _leaving_ our wallet. I think we should also have a field called `fundingAmount` which is the amount of money we are inputting into this tx. We should be able to have an invariant then that looks something like this: 

`require(fundingAmount =  txFee + outgoingAmount + changeAmount)` I think? 

Also, this does not support batching, which i'm fine with, but we should file an issue for it at minimum.

Reply:
This doesn't break with batching but breaks if we don't own every input, this would break with DLCs.